service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusServiceIT.java

```diff
@@ -54,14 +60,44 @@ public class EhrExtractStatusServiceIT {
     @Autowired
     private EhrExtractStatusRepository ehrExtractStatusRepository;
 
+    @Autowired
+    private EhrExtractTimeoutScheduler ehrExtractTimeoutScheduler;
+
     @MockBean
     private TimestampService timestampService;
 
+    @MockBean
+    private Logger logger;
+
     @BeforeEach
     public void emptyDatabase() {
         ehrExtractStatusRepository.deleteAll();
     }
 
+    @Test
+    void updateEhrExtractStatusAckTest() {
+        var inProgressConversationId = generateRandomUppercaseUUID();
+
+        EhrExtractStatusService ehrExtractStatusServiceSpy = spy(ehrExtractStatusService);
+
+        addInProgressTransferWithExceededAckTimeout(inProgressConversationId, List.of());
+
+        ehrExtractTimeoutScheduler.processEhrExtractAckTimeouts();
+        when(ehrExtractStatusServiceSpy.logger()).thenReturn(logger);
+
+        var ehrReceivedAcknowledgement = getEhrReceivedAcknowledgement(inProgressConversationId);
+        ehrReceivedAcknowledgement.setReceived(NOW);
+        ehrReceivedAcknowledgement.setConversationClosed(NOW);
+
+        ehrExtractStatusServiceSpy.updateEhrExtractStatusAck(inProgressConversationId, ehrReceivedAcknowledgement);
+
+        verify(logger, times(1))
+            .warn("Received an ACK message with conversation_id: {}, "
+                  + "but it is being ignored because the EhrExtract has already been marked as failed "
+                  + "from not receiving an acknowledgement from the requester in time.",
+                  inProgressConversationId);
```

Comment:
Asserting against log messages in software development can lead to several issues that can affect the reliability, maintainability, and overall effectiveness of the testing process. Here are some reasons why it is generally advisable to avoid assertions based on log messages:

### Fragility of Log Messages
* Changing Formats: Log message formats can change over time due to modifications in the logging framework, changes in the application code, or updates in logging standards. Assertions against log messages can become brittle and lead to frequent test failures.
* Localization: If an application supports multiple languages, log messages may vary based on the locale. This can result in assertions failing when they shouldn't, as the expected message might not match the actual message due to translation.
### Performance Concerns
* Overhead: Extracting and asserting log messages can introduce unnecessary overhead, especially in performance-sensitive applications. This might slow down test execution and consume additional resources.
* Logging Levels: During testing, certain log levels might be disabled (e.g., debug or info), which can lead to incomplete logs and subsequent assertion failures.
### Indirect Testing of Functionality
* Testing Side Effects: Relying on log messages may lead to a focus on verifying side effects rather than the primary functionality of the code. This can result in a situation where tests pass because log messages are correct, but the actual business logic is faulty.
* Misleading Success: If a test passes because the log message is correct but the application fails to perform its intended function, it can create a false sense of security regarding the system’s correctness.
### Clarity and Intent
* Test Readability: Tests that assert log messages can be harder to read and understand, especially for new developers or team members. Clear assertions on the expected state or output of the system are generally more straightforward and intuitive.
* Focus on Requirements: Tests should ideally assert the expected behaviour or outcomes of a system rather than internal implementation details such as log messages.
### Log Messages as Implementation Details
* Encapsulation: Log messages are often implementation details that shouldn't be part of the external behaviour contract of an application. By asserting on them, tests become tightly coupled to the implementation, which is contrary to best practices in software design.
Conclusion
Instead of asserting against log messages, it is often better to verify the outcomes or state changes that occur as a result of the code execution. This approach leads to more robust tests that are less likely to break with changes in the codebase while providing greater confidence in the application's functionality.

### Conclusion
Instead of asserting against log messages, it is often better to verify the outcomes or state changes that occur as a result of the code execution. This approach leads to more robust tests that are less likely to break with changes in the codebase while providing greater confidence in the application's functionality.

### References
* Sussman, G. J., & Steele, G. L. (1980). "Scheme: An Interpreter for Extended Lambda Calculus." Proceedings of the ACM SIGPLAN 1980 Symposium on Language Issues in Programming Environments.
* B. M. H. (2016). "Effective Unit Testing: A guide for Java developers." Springer.
* Martin, R. C. (2008). "Clean Code: A Handbook of Agile Software Craftsmanship." Prentice Hall.
* Crispin, L., & Gregory, J. (2009). "Agile Testing: A Practical Guide for Testers and Agile Teams." Addison-Wesley.

Reply:
in general and broader terms it makes sense, but in my particular case the logging with this message is the requested business logic, hence I am testing specifically for this scenario to make sure that it is triggered.